src/view/components/extensionView.jsx

```diff
@@ -26,23 +26,37 @@ const ExtensionView = ({
   validationSchema,
   render
 }) => {
-  const [{ initialized, initialValues, initInfo }, setState] = useState({
+  const [
+    { initialized, initialValues, initInfo },
+    setInitializationValues
+  ] = useState({
     initialized: false,
     initialValues: {},
     initInfo: {}
   });
 
   useEffect(() => {
+    // This effect function only runs once, so while our registered init
+    // function updates the component's state using setInitializationValues,
+    // the getSettings closure wouldn't be able to access the updated component
+    // state value. For this reason, we keep this local effectInitInfo
+    // so that getSettings() can access the initInfo value after it comes
+    // through init().
+    let effectInitInfo;
```

Comment:
This is a weird conundrum. There might be a better way to structure this component/extensionbridge relationship, but I would need to give it some deep thought.